js/packages/cli/src/commands/verifyTokenMetadata/index.ts

```diff
@@ -0,0 +1,166 @@
+  if (image !== expectedImagePath) {
+    // We _could_ match against this in the JSON schema validation, but it is totally valid to have arbitrary URLs to images here.
+    // The downside, though, is that those images will not get uploaded to Arweave since they're not on-disk.
+    log.warn(`We expected the \`image\` property in ${manifestFile} to be ${expectedImagePath}.
+This will still work properly (assuming the URL is valid!), however, this image will not get uploaded to Arweave through the \`metaplex upload\` command.   
+If you want us to take care of getting this into Arweave, make sure to set \`image\`: "${expectedImagePath}"
+The \`metaplex upload\` command will automatically substitute this URL with the Arweave URL location.
+    `);
+  }
+  const pngFiles = files.filter(file => file.type === 'image/png');
+  if (pngFiles.length === 0 || !pngFiles.some(file => file.uri === image)) {
+    throw new Error(
+      `At least one entry with the \`image/png\` type in the \`properties.files\` array is expected to match the \`image\` property.`,
+    );
+  }
+};
+
+export const verifyConsistentShares = (collatedCreators: CollatedCreators) => {
+  // We expect all creators to have been added to the same amount of tokens
+  const tokenCountSet = new Set<number>();
+  for (const [address, collation] of collatedCreators.entries()) {
+    tokenCountSet.add(collation.tokenCount);
+    if (tokenCountSet.size > 1) {
+      log.warn(
+        `We found that ${address} was added to more tokens than other creators.`,
+      );
+    }
+  }
+};
+
+export const verifyMetadataManifests = ({ files }) => {
+  const manifestFiles = files.filter(
+    file => path.extname(file) === EXTENSION_JSON,
+  );
+
+  // Used to keep track of the share allocations for individual creators
+  // We will send a warning if we notice discrepancies across the entire collection.
+  const collatedCreators: CollatedCreators = new Map();
+
+  // Do manifest-specific stuff here
+  for (const manifestFile of manifestFiles) {
+    // Check the overall schema shape. This is a non-exhaustive check, but guarantees the bare minimum needed for the rest of the commands to succeed.
+    const tokenMetadata = require(manifestFile) as TokenMetadata;
+    validate(tokenMetadata, tokenMetadataJsonSchema, { throwError: true });
+
+    const {
+      properties: { creators },
+    } = tokenMetadata;
+    verifyAggregateShare(creators, manifestFile);
+
+    verifyCreatorCollation(creators, collatedCreators, manifestFile);
+
+    // Check that the `image` and at least one of the files has a URI matching the index of this token.
+    const {
+      image,
+      properties: { files },
+    } = tokenMetadata;
+    verifyImageURL(image, files, manifestFile);
+  }
+
+  verifyConsistentShares(collatedCreators);
+};
+
+export const verifyTokenMetadata = ({
+  files,
+  uploadElementsCount = null,
+}): Boolean => {
+  // Will we need to deal with the cache?
```

Comment:
i'm not sure whether we need to take into account the state of the cache or not. my sense is "no", but i could use a second pair of eyes